src/elements_parser.js

```diff
@@ -72,7 +72,19 @@ fabric.ElementsParser = function(elements, callback, options, reviver, parsingOp
   proto.resolveGradient = function(obj, property) {
     var gradientDef = this.extractPropertyDefinition(obj, property, 'gradientDefs');
     if (gradientDef) {
-      obj.set(property, fabric.Gradient.fromElement(gradientDef, obj));
+      var gradient = fabric.Gradient.fromElement(gradientDef, obj);
```

Comment:
Ok this is good, but i would like to have it differently.
resolveGradient is called from createCallback, where `el` the original element is available.

I would like to have resolveGradient taking the `el` as additional argument, extract el.getAttribute('fill-opacity') in this function, and have this multiplication logic in the `fromElement` itself.

All in all would be less code.
We could extract fillOpacity inside Gradient.fromElement since we pass the obj in the function, but the point is that obj is already a fabric instance and fillOpacity, while collected during svg parsing, is not an object property and should be removed one day.

I think the solution is good, thank you for finding it.

Reply:
I think this didn't turn out to be less code after all...

Reply:
can t you bring the opacity value inside gradient.fromElement? Wasn't that doable?

Reply:
I don't think so, since the fill-opacity is on the path element, not the gradient element.

Reply:
Oh.  Duh.